clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -69,14 +70,30 @@ public Cluster(String clusterId,
                    Set<String> unauthorizedTopics,
                    Set<String> internalTopics,
                    Node controller) {
-        this(clusterId, false, nodes, partitions, unauthorizedTopics, internalTopics, controller);
+        this(clusterId, false, nodes, partitions, unauthorizedTopics, Collections.<String>emptySet(), internalTopics, controller);
+    }
+
+    /**
+     * Create a new cluster with the given id, nodes and partitions
+     * @param nodes The nodes in the cluster
+     * @param partitions Information about a subset of the topic-partitions this cluster hosts
+     */
+    public Cluster(String clusterId,
+        Collection<Node> nodes,
+        Collection<PartitionInfo> partitions,
+        Set<String> unauthorizedTopics,
+        Set<String> invalidTopics,
+        Set<String> internalTopics,
+        Node controller) {
```

Comment:
It's a bit awkward to keep adding parameters. Another option would be to have a Map from error to the set of topics.

Reply:
I agree with your sentiment. However, I feel in this case adding an additional parameter is fine for the following reason:
- Stating the parameters separately explicitly conveys to the reader the meaning of the different sets of topics.  Putting the set of topics into a map<string, Set<String>> would loose that value/visibility.
- If the set of topics were put into a map, the map would contain only three items.  For me, that's not an enough of a number of items in the map to justify using a map.  And I don't see the number of set of topics passed as a parameter increasing in the near future.

Reply:
Seems OK for now.